orttraining/orttraining/eager/ort_aten.cpp

```diff
@@ -1072,6 +1072,81 @@ at::Tensor& _log_softmax_out(
   return out;
 }
 
+// aten::mm.out(Tensor self, Tensor mat2, *, Tensor(a!) out) -> Tensor(a!)
+// mm is for matrix multiplication and does not broadcast.
+// https://pytorch.org/docs/stable/generated/torch.mm.html
+at::Tensor& mm_out(
+  const at::Tensor& self,
+  const at::Tensor& mat2,
+  // *,
+  at::Tensor& out) {
+  ORT_LOG_FN(self, mat2, out);
+
+  if (
+    std::vector<at::ScalarType> supportedTypes =
+      {at::kDouble, at::kLong, at::kHalf, at::kFloat, at::kBFloat16, at::kInt};
+    !IsSupportedType(self, supportedTypes) ||
+    !IsSupportedType(mat2, supportedTypes)) {
+    return at::native::call_fallback_fn<
+      &at::native::cpu_fallback,
+      ATEN_OP(mm_out)>::call(self, mat2, out);
+  }
+  auto& invoker = GetORTInvoker(self.device());
+
+  auto promoted_type = PromoteScalarTypesWithCategory({self.scalar_type(),mat2.scalar_type()}, {});
+
+  // to match cpu device behavior for torch.mm, we verify the following in order and copied messaged from cpu output
```

Comment:
Interesting, so pytorch doesn't verify these before dispatching?
What about falling back to CPU in these cases to allow the checks/errors to come from the existing CPU functions instead of recreating them?

Reply:
i like that idea of falling back to cpu. I'll give that a try. Thanks for the suggestion.

Reply:
@msftlincoln - thanks for the suggestion again. I have changed it to fallback and keep the error message on the cpu side.